### PR TITLE
ISSUE-678 JMAP - EmailRecoveryAction/set create

### PR DESCRIPTION
**Pull request by vttranlina**
resolve https://github.com/linagora/tmail-backend/issues/678

### CODE REVIEW

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/EmailRecoveryActionSetMethodContract.scala
Comment (Arsnael) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/EmailRecoveryActionSetMethodContract.scala:63]: debug spotted, dont forget it later :)

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/CustomMethod.scala
Comment (Arsnael) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/CustomMethod.scala:26]: How about `com:linagora:params:jmap:messages:vault`

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/EmailRecoveryActionSetMethod.scala
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/EmailRecoveryActionSetMethod.scala:116]: This blocking call should be wrapped...

**Comment by vttranlina** [comment]
TODO

- [ ] Adapt James make public some methods & RestoreService should support query `limit`

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/EmailRecoveryActionSetMethodContract.scala
Comment (chibenwa) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/EmailRecoveryActionSetMethodContract.scala:242]:
```suggestion
    `given`()
      .spec(webAdminApi)
      .get(taskId)
    .`then`()
      .statusCode(SC_OK)
      .body("taskId", Matchers.is(taskId))
```

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/EmailRecoveryActionSetMethodContract.scala
Comment (chibenwa) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/EmailRecoveryActionSetMethodContract.scala:1523]:
```suggestion
  def awaitRestoreTaskCompleted(taskId: String): ValidatableResponse =
```

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/EmailRecoveryActionSetMethodContract.scala
Comment (chibenwa) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/EmailRecoveryActionSetMethodContract.scala:1537]:
```suggestion
    await().atMost(30, TimeUnit.SECONDS).untilAsserted(() => assertThat(listAllMessageResult(guiceJamesServer, username)).hasSize(size))
```

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/EmailRecoveryActionSetMethodContract.scala
Comment (chibenwa) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/EmailRecoveryActionSetMethodContract.scala:1534]:
```suggestion
  def listAllMessageResult(guiceJamesServer: GuiceJamesServer, username: Username): java.util.Collection[MessageId] =
    guiceJamesServer.getProbe(classOf[MailboxProbeImpl])
      .searchMessage(MultimailboxesSearchQuery.from(SearchQuery.of(SearchQuery.all())).build, username.asString(), 100)
```

**Comment by chibenwa** [comment]
> Adapt James make public some methods & RestoreService should support query limit

Separate PR please

**Review by Arsnael — COMMENTED** [review]
Read it, nothing to add, good job :)

**Comment by vttranlina** [comment]
> Adapt James make public some methods & RestoreService should support query limit

https://github.com/apache/james-project/pull/1572

**Comment by vttranlina** [comment]
> ```
> [ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:4.6.1:compile (scala-compile-first) on project jmap-extensions: Execution scala-compile-first of goal net.alchim31.maven:scala-maven-plugin:4.6.1:compile failed: Compilation failed: InterfaceCompileFailed -> [Help 1]
> ```

we wait until the commit was merged first https://github.com/apache/james-project/pull/1572/commits/4efb8f20c80cb8823bc0ce4705607e63508ece3b , then update james to tmail, and rebase this pr

**Review by quantranhong1999 — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by vttranlina** [comment]
rebase master & squash

**Pull request merged**
